### ISSUE TITLE
Discover and Series issue

### ISSUE THREAD

**Issue by programatix** [Flexget/Flexget#1163]
I'm having issue where whenever I added new items into _series_, _discover_ does not include them. The following is a partial sample of my config:
```
templates:
  cartoons:
    series:
      settings:
        anime:
          propers: 3 days
          quality: <=720p
          identified_by: sequence
          sequence_regexp:
            - (\d+)
            - S(\d+) (\d+)
          from_group: horriblesubs
          set:
            path: '/mnt/storage1/Videos/TV Series/Various/{{series_name}}'
      anime:
        - Kuromukuro:
            from_group: DeadFish
        - Bungou Stray Dogs 
        - Big Order

tasks:
  NyaaTorrents:
    template: [cartoons]
    discover:
      what:
        - emit_series:
            backfill: no
            from_start: yes
      from:
        - nyaa: anime
      interval: 1 hour
```
Executing the task _NyaaTorrents_ will display the following log,
```
[date] [time] VERBOSE  discover      NyaaTorrents    Discovering 3 titles ...
```
follow up with the correct list being searched.

Now, say we add 2 new items like this,
```
templates:
  cartoons:
    series:
      settings:
        anime:
          propers: 3 days
          quality: <=720p
          identified_by: sequence
          sequence_regexp:
            - (\d+)
            - S(\d+) (\d+)
          from_group: horriblesubs
          set:
            path: '/mnt/storage1/Videos/TV Series/Various/{{series_name}}'
      anime:
        - Kuromukuro:
            from_group: DeadFish
        - Bungou Stray Dogs 
        - Big Order
        - Hundred
        - Endride

tasks:
  NyaaTorrents:
    template: [cartoons]
    discover:
      what:
        - emit_series:
            backfill: no
            from_start: yes
      from:
        - nyaa: anime
      interval: 1 hour
```
Executing the task _NyaaTorrents_ now (with or without --no-cache) will show the following log
```
[date] [time] VERBOSE  discover      NyaaTorrents    Discovering 3 titles ...
```
and the list being search is the original 3 items. The 2 new items will not appear.

However, if I edit the existing list, for example, add _from_group_ to refer to other group, it will be executed correctly for the original items. Eg,
```
        - Kuromukuro:
            from_group: DeadFish
        - Bungou Stray Dogs          # this updated item will be searched correctly
            from_group: DeadFish
        - Big Order
```

**Comment by Jonybat** [comment]
Looks like the same issue as i described in #957

**Comment by cvium** [comment]
Fixed by 7c29d3e925a5729a30e59f40f2fae009ea8aa4b6
